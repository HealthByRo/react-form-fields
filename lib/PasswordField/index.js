'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _PasswordInput = require('../PasswordInput');

var _PasswordInput2 = _interopRequireDefault(_PasswordInput);

var _createFormField = require('../createFormField');

var _createFormField2 = _interopRequireDefault(_createFormField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _createFormField2.default)(_PasswordInput2.default);