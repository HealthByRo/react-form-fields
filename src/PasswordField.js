/* @flow */

import React, { PureComponent } from 'react';
import { LabelElement, PasswordFieldElement } from './elements';
import { FieldWrapper } from './wrappers';
import Errors from './Errors';
import type { FieldProps } from './types';

export default class PasswordField extends PureComponent {
  constructor(props: FieldProps) {
    super(props);

    if (!props.name || props.name.length < 1) {
      // TODO create file with errors, add MissingParamError
      throw new Error('name param is required for PasswordField');
    }
  }

  composeProps(): FieldProps {
    const props = { ...this.props };

    if (!props.inputId) {
      props.inputId = this.generateInputId(props);
    }

    return props;
  }

  // TODO add some unique id at the end
  generateInputId(props: FieldProps): string {
    return `TF-${props.name}`;
  }

  props: FieldProps;

  render() {
    const props = this.composeProps();

    return (
      <FieldWrapper>
        {props.label &&
          <LabelElement {...props} />
        }
        <PasswordFieldElement {...props} />
        <Errors {...this.props} />
      </FieldWrapper>
    );
  }
}
