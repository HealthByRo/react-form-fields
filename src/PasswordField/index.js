/* @flow */

import React from 'react';
import { composeInputProps } from '../helpers';
import type { InputProps, TextFieldElementProps } from '../types';

const PasswordField = (props: TextFieldElementProps) => {
  const inputProps: InputProps = composeInputProps('password', props);

  return (
    <input type="password" {...inputProps} />
  );
};
// TODO figure out a better way of using defaults in stateless component
PasswordField.defaultProps = {
  className: 'textfield',
};

export default PasswordField;
