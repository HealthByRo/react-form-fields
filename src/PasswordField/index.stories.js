import React from 'react';
import { storiesOf } from '@kadira/storybook';
import PasswordField from './';

storiesOf('Fields: PasswordField', module)
  .add('input', () => {
    const props = {
      name: 'PASSWORD_FIELD_NAME',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + placeholder', () => {
    const props = {
      name: 'PASSWORD_FIELD_NAME',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'PASSWORD_FIELD_NAME',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + placeholder + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'PASSWORD_FIELD_NAME',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label', () => {
    const props = {
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label + multiple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label + placeholder', () => {
    const props = {
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label + placeholder + error', () => {
    const props = {
      errors: ['Sample error for field'],
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  })
  .add('input + label + placeholder + mutliple errors', () => {
    const props = {
      errors: [
        'First error for field',
        'Second error for field',
      ],
      name: 'PASSWORD_FIELD_NAME',
      label: 'Sample label',
      placeholder: 'Sample password field with placeholder',
    };

    return (
      <PasswordField {...props} />
    );
  });
