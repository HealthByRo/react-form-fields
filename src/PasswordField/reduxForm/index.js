import PasswordField from '../';
import createReduxFormField from '../../createReduxFormField';

export default createReduxFormField(PasswordField);
