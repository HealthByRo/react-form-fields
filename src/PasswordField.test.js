/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import PasswordField from './PasswordField';
import type { FieldProps } from './types';

let tree;
const placeholder = 'INPUT_PLACEHOLDER';
const label = 'INPUT_LABEL';
const renderWithProps = (props: FieldProps) => {
  tree = renderer.create(
    <PasswordField {...props} />
  ).toJSON();
};
const getPropsWithName = (props?: Object, name?: string = 'INPUT_NAME'): FieldProps => {
  const propsWithName = { ...props };

  if (!propsWithName.name) {
    propsWithName.name = name;
  }

  return propsWithName;
};

describe('<PasswordField />', () => {
  it('should throw error if no name passed', () => {
    // $FlowExpectedError
    expect(() => renderWithProps()).toThrowError('name param is required for PasswordField');
  });
});

describe('<PasswordField { name } />', () => {
  it('should NOT throw any errors if name is passed', () => {
    expect(() => <PasswordField name="TEST_NAME" />).not.toThrowError();
  });

  it('should render only wrapped input field with defaults', () => {
    renderWithProps(getPropsWithName());

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, placeholder } />', () => {
  it('should render input with placeholder', () => {
    renderWithProps(getPropsWithName({ placeholder }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, placeholder, errors: [SINGLE_ERROR] } />', () => {
  it('should render input with placeholder and single error', () => {
    renderWithProps(getPropsWithName({ placeholder, errors: ['SINGLE_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, placeholder, errors: [FIRST_ERROR, SECOND_ERROR] } />', () => {
  it('should render input with placeholder and two error', () => {
    renderWithProps(getPropsWithName({ placeholder, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label } />', () => {
  it('should render input and label', () => {
    renderWithProps(getPropsWithName({ label }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label, errors: [SINGLE_ERROR] } />', () => {
  it('should render input and label and single error', () => {
    renderWithProps(getPropsWithName({ label, errors: ['SINGLE_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label, errors: [FIRST_ERROR, SECOND_ERROR] } />', () => {
  it('should render input and label and two error', () => {
    renderWithProps(getPropsWithName({ label, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label, placeholder } />', () => {
  it('should render input with placeholder and label', () => {
    renderWithProps(getPropsWithName({ label, placeholder }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label, placeholder, errors: [SINGLE_ERROR] } />', () => {
  it('should render input with placeholder, label and single error', () => {
    renderWithProps(getPropsWithName({ label, placeholder, errors: ['SINGLE_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});

describe('<PasswordField { name, label, placeholder, errors: [FIRST_ERROR, SECOND_ERROR]  } />', () => {
  it('should render input with placeholder, label and two errors', () => {
    renderWithProps(getPropsWithName({ label, placeholder, errors: ['FIRST_ERROR', 'SECOND_ERROR'] }));

    expect(tree).toMatchSnapshot();
  });
});
