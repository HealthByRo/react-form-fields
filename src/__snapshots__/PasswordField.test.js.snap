// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> multiple params name + label + errors[1] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="password"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + label + errors[2] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="password"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + label + placeholder + errors[1] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + label + placeholder + errors[2] 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + label + placeholder 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
</div>
`;

exports[`<PasswordField /> multiple params name + label 1`] = `
<div
  className="form-item"
>
  <label
    htmlFor="TF-INPUT_NAME"
  >
    INPUT_LABEL
  </label>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="password"
  />
</div>
`;

exports[`<PasswordField /> multiple params name + placeholder + errors[1] 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
  <div
    className="msg msg--error"
  >
    SINGLE_ERROR
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + placeholder + errors[2] 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
  <div
    className="errors-list"
  >
    <div
      className="msg msg--error"
    >
      FIRST_ERROR
    </div>
    <div
      className="msg msg--error"
    >
      SECOND_ERROR
    </div>
  </div>
</div>
`;

exports[`<PasswordField /> multiple params name + placeholder 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    placeholder="INPUT_PLACEHOLDER"
    type="password"
  />
</div>
`;

exports[`<PasswordField /> param name: should render only wrapped input field with defaults 1`] = `
<div
  className="form-item"
>
  <input
    className="textfield"
    id="TF-INPUT_NAME"
    name="INPUT_NAME"
    type="password"
  />
</div>
`;
